node/src/components/block_proposer/event.rs

```diff
@@ -14,24 +14,59 @@ use crate::{
     types::{DeployHash, DeployHeader, ProtoBlock},
     Chainspec,
 };
+use casper_execution_engine::shared::motes::Motes;
 
 /// A wrapper over `DeployHeader` to differentiate between wasm-less transfers and wasm headers.
 #[derive(Clone, DataSize, Debug, Deserialize, Serialize)]
 pub enum DeployType {
     /// Represents a wasm-less transfer.
-    Transfer(DeployHeader),
+    Transfer {
+        header: DeployHeader,
+        payment_amount: Motes,
+        size: usize,
+    },
     /// Represents a wasm deploy.
-    Wasm(DeployHeader),
+    Other {
+        header: DeployHeader,
+        payment_amount: Motes,
+        size: usize,
+    },
```

Comment:
(It looks like this shouldn't be an enum at all: Both variants have the same fields. But it's a matter of taste.)

Reply:
For now I'd like to leave as is, but if this ends up remaining the case for long, I'd be ok to refactor this as you suggest.